lib/tokenizer.ts

```diff
@@ -1,43 +1,40 @@
-const util      = require('./util');
-const types     = require('./types');
-const sets      = require('./sets');
-const positions = require('./positions');
-
-
-module.exports = (regexpStr) => {
-  let i = 0, l, c;
-  let start = { type: types.ROOT, stack: []};
+import * as util from './util'
+import { Group, types, Root, Token } from './types'
+import * as sets from './sets'
+
+/**
+ * Tokenizes a regular expression (that is currently a string)
+ * @param regexpStr String of regular expression to be tokenized
+ */
+export const tokenizer = (regexpStr: string): Root => {
+  let i: number = 0, c: string;
+  let start: Root = { type: types.ROOT, stack: [] };
 
   // Keep track of last clause/group and stack.
-  let lastGroup = start;
-  let last = start.stack;
-  let groupStack = [];
+  let lastGroup: Group | Root = start;
+  let last: Token[] = start.stack ?? [];
```

Comment:
`start.stack` should always be defined here
```suggestion
  let last: Token[] = start.stack;
```